src/main/java/frc/robot/commands/TeleopSwerve.java

```diff
@@ -0,0 +1,93 @@
+package frc.robot.commands;
+
+import frc.robot.Constants;
+import frc.robot.States;
+import frc.robot.subsystems.swerve.Swerve;
+import frc.robot.subsystems.swerve.SwerveConfig;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.Command;
+
+
+public class TeleopSwerve extends Command {    
+    private Swerve s_Swerve;    
+    private DoubleSupplier translationSup;
+    private DoubleSupplier strafeSup;
+    private DoubleSupplier rotationSup;
+    private BooleanSupplier robotCentricSup;
+    private BooleanSupplier dampen;
+    private DoubleSupplier speedDial;
+
+    private PIDController rotationController;
+
+    public TeleopSwerve(Swerve s_Swerve, DoubleSupplier translationSup, DoubleSupplier strafeSup, DoubleSupplier rotationSup, BooleanSupplier robotCentricSup, BooleanSupplier dampen, DoubleSupplier speedDial) {
+        this.s_Swerve = s_Swerve;
+        addRequirements(s_Swerve);
+
+        rotationController = new PIDController(SwerveConfig.angleKP, SwerveConfig.angleKI, SwerveConfig.angleKD);
+        rotationController.enableContinuousInput(-Math.PI, Math.PI);
+        rotationController.setTolerance(0.1);
+
+        this.translationSup = translationSup;
+        this.strafeSup = strafeSup;
+        this.rotationSup = rotationSup;
+        this.robotCentricSup = robotCentricSup;
+        this.dampen = dampen;
+        this.speedDial = speedDial;
+    }
+
+    @Override
+    public void execute() {
+        /* Get Values, Deadband*/
+        double translationVal = MathUtil.applyDeadband(translationSup.getAsDouble(), Constants.stickDeadband) * (dampen.getAsBoolean() ? 0.2 : 1) * ((speedDial.getAsDouble() + 1) / 2);
```

Comment:
For clarity, calculate the dampen value once, and store it in a variable.  Then multiply that variable by each deadband-corrected value as needed.

Reply:
Will do, I'll do the same for speedDial as well.